### PR TITLE
fix: set GO usage environment variables from resource limits

### PR DESCRIPTION
**Pull request by rogercoll**

### CODE REVIEW

**Comment by rogercoll** [comment]
All variables remain and GOMAXPROCS value is rounded up:

```
Environment:
      OTEL_K8S_NODE_NAME:   (v1:spec.nodeName)
      OTEL_K8S_NODE_IP:     (v1:status.hostIP)
      OTEL_K8S_NAMESPACE:  opentelemetry-operator-system (v1:metadata.namespace)
      OTEL_K8S_POD_NAME:   opentelemetry-kube-stack-daemon-collector-9brfc (v1:metadata.name)
      OTEL_K8S_POD_IP:      (v1:status.podIP)
      ELASTIC_AGENT_OTEL:  true
      GOMEMLIMIT:          1025MiB
      ELASTIC_API_KEY:     <set to the key 'elastic_api_key' in secret 'elastic-secret-otel'>   Optional: false
      ELASTIC_ENDPOINT:    <set to the key 'elastic_endpoint' in secret 'elastic-secret-otel'>  Optional: false
      GOMAXPROCS:          2 (limits.cpu)
```

File: resources/kubernetes/operator/helm/values.yaml [hunk 1]
```diff
@@ -219,11 +219,30 @@ collectors:
...
+        valueFrom:
+          secretKeyRef:
+            name: elastic-secret-otel
+            key: elastic_api_key
+      - name: GOMAXPROCS
+        valueFrom:
+          resourceFieldRef:
+            resource: limits.cpu
+      - name: GOMEMLIMIT
+        value: "1025MiB"
```
Comment (lahsivjar) [inline resources/kubernetes/operator/helm/values.yaml:245]: Maybe we can write a script to keep this 90% (or lower if we want to go lower) of the memory limit? Will be a bit easier to manage and update in future.

**Review by lahsivjar — APPROVED** [review]
LGTM! One nit regarding hard coded mem limit.

File: resources/kubernetes/operator/helm/values.yaml (hunk 1 above)
Comment (rogercoll) [reply resources/kubernetes/operator/helm/values.yaml:245]: Definitely, that would be a better setting. As discussed, we can help with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30289 moving it forward.

**Pull request merged**